### ISSUE TITLE
BaseJUnit4IteratorTest tests fail to run 1.10

### ISSUE THREAD

**Issue by ctubbsii** [apache/accumulo#1682]
I noticed while working on #1681 that the SummingCombinerTest, and other tests which use the iterator-test-harness via BaseJUnit4IteratorTest are not actually executing. JUnit tries to run them, but fails to detect any actual test cases within them to run.

These same tests run on the main branch, so there must be something different about 1.x (possibly the version of JUnit, but upgrading that would introduce a bunch of warnings that would probably need to be addressed, and which have already been addressed in the main branch).

**Comment by ctubbsii** [comment]
I tried bumping to JUnit 4.13, and it didn't help. The tests are not found in those classes when run on 1.10 branch, but are found fine in main branch. The test classes themselves are nearly identical, and their differences wouldn't explain this behavior. Very strange.

**Comment by dlmarion** [comment]
I can try to replicate the issue you are having.

**Comment by ctubbsii** [comment]
The command I ran was:

`mvn clean package -Dtest=BaseJUnit4IteratorTest,JUnitFrameworkTest,AgeOffFilterTest,CfCqSliceFilterTest,SummingCombinerTest,WholeRowIteratorTest`

If you follow the output from Maven, you can see on 1.10, it shows something like:
```
[INFO] --- maven-surefire-plugin:3.0.0-M3:test (default-test) @ accumulo-test ---
[INFO] 
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running org.apache.accumulo.test.iterator.WholeRowIteratorTest
[INFO] Tests run: 0, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.308 s - in org.apache.accumulo.test.iterator.WholeRowIteratorTest
[INFO] Running org.apache.accumulo.test.iterator.CfCqSliceFilterTest
[INFO] Tests run: 0, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.325 s - in org.apache.accumulo.test.iterator.CfCqSliceFilterTest
[INFO] Running org.apache.accumulo.test.iterator.AgeOffFilterTest
[INFO] Tests run: 0, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.37 s - in org.apache.accumulo.test.iterator.AgeOffFilterTest
[INFO] Running org.apache.accumulo.test.iterator.SummingCombinerTest
[INFO] Tests run: 0, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.377 s - in org.apache.accumulo.test.iterator.SummingCombinerTest
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 0, Failures: 0, Errors: 0, Skipped: 0
```

However, on main, it shows something like:
```
[INFO] --- maven-surefire-plugin:3.0.0-M3:test (default-test) @ accumulo-test ---
[INFO] 
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running org.apache.accumulo.test.iterator.WholeRowIteratorTest
[INFO] Tests run: 6, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 1.904 s - in org.apache.accumulo.test.iterator.WholeRowIteratorTest
[INFO] Running org.apache.accumulo.test.iterator.CfCqSliceFilterTest
[INFO] Tests run: 6, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 2.028 s - in org.apache.accumulo.test.iterator.CfCqSliceFilterTest
[INFO] Running org.apache.accumulo.test.iterator.AgeOffFilterTest
[INFO] Tests run: 6, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 1.881 s - in org.apache.accumulo.test.iterator.AgeOffFilterTest
[INFO] Running org.apache.accumulo.test.iterator.SummingCombinerTest
[INFO] Tests run: 6, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 1.962 s - in org.apache.accumulo.test.iterator.SummingCombinerTest
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 24, Failures: 0, Errors: 0, Skipped: 0
```

So, surefire is finding the test, and running it with JUnit, but JUnit isn't finding the test cases in 1.10, for whatever reason.

**Comment by ctubbsii** [comment]
I believe @dlmarion identified the bug being with the older version of Guava used to find tests, which looks for classes under the assumption that the classloader is of URLClassLoader type, which it isn't in Java 11. The tests were confirmed to pass using Java 8, and these tests pass fine in the main branch. Does this description, match your findings @dlmarion ?

It may be possible to fix this in the 1.10 branch, but it's probably not worth it. I'm closing as "won't fix".

**Comment by dlmarion** [comment]
That is correct.
